### DISCUSSION TITLE
cannot compare refreshClaims.ExpiresAt < time.Now().Unix() (operator < not defined for *jwt.NumericDate)

### DISCUSSION THREAD

**Discussion by raliqala** [golang-jwt/jwt#145] [Q&A] [answered]
Hello 

Good day  

specs  

- Windows 11
- Golang latest version
- Fiber
- https://github.com/golang-jwt/jwt v4
- errors

1.  cannot compare refreshClaims.ExpiresAt < time.Now().Unix() (operator < not defined for *jwt.NumericDate) 
2. cannot compare refreshClaims.ExpiresAt < currentDate (operator < not defined for *jwt.NumericDate) 
3. cannot compare refreshClaims.ExpiresAt < time.Now() (operator < not defined for *jwt.NumericDate) 

Am trying to verify expiry date but since I migrated from https://github.com/dgrijalva/jwt-go to https://github.com/golang-jwt/jwt I can't compare the expiry date with the current datetime. Here is my code sample 

```
     OLD CODE
	if token.Valid {
		if refreshClaims.ExpiresAt < time.Now().Unix() {
			// refresh token is expired
			c.ClearCookie("access_token", "refresh_token")
			return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
				"success": false,
				"general": "Access Denied",
			})
		}
	}


    NEW CODE I HAVE TRIED but i still get the error
	currentDate := jwt.NewNumericDate(time.Now())

	if token.Valid {
		if refreshClaims.ExpiresAt < currentDate {
			// refresh token is expired
			c.ClearCookie("access_token", "refresh_token")
			return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
				"success": false,
				"general": "Access Denied",
			})
		}
	}
```

Please help  
Thanks in advance, Happy christmas ❄️🎄🎄❄️🎉🎁🎀🎇🎊

**Comment by oxisto** [comment, marked as answer]
Unfortunately Go does not allow to overload operators, so you cannot use `>` to compare the `jwt.NumericDate` with seconds or another time. You can however use the function `VerifyExpiresAt` which will abstract this for you:
```
currentDate = time.Now()
if !refreshClaims.VerifyExpiresAt(currentDate, true) {
 /* token is expired */
}
```

Note: in your example, the first `if token.Valid` should probably be `if !token.Valid` because if the token is valid, it can never be expired (in our current system, unless you have override this behaviour somehow).

**Reply by raliqala** [reply]
Thanks you so mach
